### PR TITLE
Backend for verifying login status for a user. 

### PR DESCRIPTION
**Pull request by MikolJiron**
Create a new Java servlet that responds with the login status of the user and the appropriate URL to either login or logout.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    UserService userService = UserServiceFactory.getUserService();
+    UserStatus status;
+    
+    // Create UserStatus based on whether the user is logged in or not.
+    if (userService.isUserLoggedIn()) {
+      // The user IS logged in.
+      String urlToRedirectToAfterUserLogsOut = "/";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+      status = new UserStatus(true, logoutUrl);
+
```
Comment (hzimz) [inline portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java:45]: nit: here and elsewhere, extra newlines.

File: portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    } else {
+      // The user IS NOT logged in.
+      String urlToRedirectToAfterUserLogsIn = "/";
```
Comment (hzimz) [inline portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java:48]: Is the plan for this and line 42 to diverge at some point? If not, could consolidate into one variable.

File: portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json;");
+    UserService userService = UserServiceFactory.getUserService();
+    UserStatus status;
+    
+    // Create UserStatus based on whether the user is logged in or not.
+    if (userService.isUserLoggedIn()) {
+      // The user IS logged in.
+      String urlToRedirectToAfterUserLogsOut = "/";
```
Comment (tinagrasso) [inline portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java:42]: new constant for base path

File: portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that sends authentication status and associated login/logout links to the client.*/
+@WebServlet("/authenticate-user")
+public class AuthenticationServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json;");
```
Comment (tinagrasso) [inline portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java:35]: constant.

File: portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    UserService userService = UserServiceFactory.getUserService();
+    UserStatus status;
+    
+    // Create UserStatus based on whether the user is logged in or not.
+    if (userService.isUserLoggedIn()) {
+      // The user IS logged in.
+      String urlToRedirectToAfterUserLogsOut = "/";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+      status = new UserStatus(true, logoutUrl);
```
Comment (tinagrasso) [inline portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java:44]: Add comment for constant variables.

```
new UserStatus(/** varName= */ true, logoutUrl);
```

File: portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType(Params.JSON_CONTENT_TYPE);
+    UserService userService = UserServiceFactory.getUserService();
+    UserStatus status;
+
+    // Create UserStatus based on whether the user is logged in or not.
+    if (userService.isUserLoggedIn()) {
+      // The user IS logged in.
+      String logoutUrl = userService.createLogoutURL(Params.BASE_URL_PATH);
+      status = new UserStatus(/** isLoggedIn= */ true, logoutUrl);
```
Comment (tinagrasso) [inline portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java:44]: If isLoggedIn is only ever going to be true or false, consider changing the input to an enum that is defined inside of UserStatus.

```
class UserStatus {
  static enum AuthState {
    LOGGED_IN = 0,
    LOGGED_OUT = 1,
  }
}
```

File: portfolio/src/main/java/com/google/sps/data/UserStatus.java [hunk 7]
```diff
@@ -17,10 +17,25 @@
...
+      this.state = state;
+    }
+
+    public Boolean getState() {
+      return this.state;
+    }
+  }
+
+  public UserStatus(Boolean isLoggedIn, String loginLogoutURL) {
     this.isLoggedIn = isLoggedIn;
```
Comment (tinagrasso) [inline portfolio/src/main/java/com/google/sps/data/UserStatus.java:39]: This should be your enum, not a boolean.

File: portfolio/src/main/java/com/google/sps/data/UserStatus.java [hunk 8]
```diff
@@ -17,10 +17,25 @@
 /** A UserStatus object to be converted to JSON and sent to the client. */
 public final class UserStatus {
 
-  private final int isLoggedIn;
+  private final Boolean isLoggedIn;
   private final String loginLogoutURL;
 
-  public UserStatus(int isLoggedIn, String loginLogoutURL) {
+  public static enum AuthState {
```
Comment (tinagrasso) [inline portfolio/src/main/java/com/google/sps/data/UserStatus.java:23]: Either move this above your instance variables or below the constructor - either make this the first thing in UserStatus or the last thing.

**Pull request merged**